app/src/main/java/com/pwhs/quickmem/presentation/app/search/SearchViewModel.kt

```
@@ -19,27 +29,75 @@ class SearchViewModel @Inject constructor() : ViewModel() {
-        } else {
-            _uiState.update {
-                it.copy(error = "Query cannot be empty")
+        }
+    }
+
+    private fun loadSearchHistory() {
+        viewModelScope.launch {
+            try {
+                val searches = searchQueryRepository.getRecentSearches()
+                _uiState.update { it.copy(listResult = searches) }
+            } catch (e: Exception) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)